Spigot-Server-Patches/0334-Adding-alternate-and-more-even-spawn-mechanic.patch

```diff
@@ -0,0 +1,547 @@
+From ed6018ece43a3ef38930a341ef38b4d15bd1966e Mon Sep 17 00:00:00 2001
+From: Colin Godsey <crgodsey@gmail.com>
+Date: Fri, 10 Aug 2018 19:01:16 -0600
+Subject: [PATCH] Adding alternate and more even spawn mechanic
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+index 61c8b58b1..b3cdee58b 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+@@ -355,6 +355,14 @@ public class PaperWorldConfig {
+         armorStandEntityLookups = getBoolean("armor-stands-do-collision-entity-lookups", true);
+     }
+ 
++    public boolean useImprovedSpawning = true;
++    private void useImprovedSpawning() {
++        useImprovedSpawning = getBoolean("use-improved-spawning", false);
++        if (useImprovedSpawning) {
++            log("Using improved natural mob spawning.");
++        }
++    }
++
+     public int maxCollisionsPerEntity;
+     private void maxEntityCollision() {
+         maxCollisionsPerEntity = getInt( "max-entity-collisions", this.spigotConfig.getInt("max-entity-collisions", 8) );
+diff --git a/src/main/java/net/minecraft/server/BiomeBase.java b/src/main/java/net/minecraft/server/BiomeBase.java
+index 1a468e1bb..b3b1b0395 100644
+--- a/src/main/java/net/minecraft/server/BiomeBase.java
++++ b/src/main/java/net/minecraft/server/BiomeBase.java
+@@ -125,7 +125,7 @@ public abstract class BiomeBase {
+     private final Map<EnumCreatureType, List<BiomeBase.BiomeMeta>> ba = Maps.newHashMap();
+ 
+     public static int a(BiomeBase biomebase) {
+-        return BiomeBase.REGISTRY_ID.a((Object) biomebase);
```

Comment:
this and the next two changes were just bad decompiles... not sure what to do with these